### PR TITLE
DIRECTOR: LINGO: Implement ARRAY variable

### PR DESCRIPTION
**Pull request by rvanlaar**
Arrays can be:
- assigned to variables and 
- those variables can be fetched.

Includes implementation to `put`, i.e. print the array.

### CODE REVIEW

File: engines/director/lingo/lingo-code.cpp [hunk 1]
```diff
@@ -226,6 +226,9 @@ void LC::c_printtop(void) {
 	case OBJECT:
 		warning("#%s", d.u.s->c_str());
 		break;
+	case ARRAY:
+		warning("ARRAY: %s", d.toString()->c_str());
```
Comment (sev-) [inline engines/director/lingo/lingo-code.cpp]: That should be just toString(). The thing is that I am not sure that I implemented toString() for arrays correctly. Somebody needs to test that. And here it should produce the output as it is now, e.g. "[<el1>, <el2>. ..]"

File: engines/director/lingo/lingo-codegen.cpp [hunk 2]
```diff
@@ -527,6 +527,8 @@ void Lingo::varAssign(Datum &var, Datum &value) {
 			sym->u.s = value.u.s;
 		} else if (value.type == VOID) {
 			sym->u.i = 0;
+		} else if (value.type == ARRAY) {
+			sym->u.farr = new DatumArray(*value.u.farr);
```
Comment (sev-) [inline engines/director/lingo/lingo-codegen.cpp]: are you sure we need to copy the whole thing and not just pass the reference here?

File: engines/director/lingo/lingo-code.cpp (hunk 1 above)
Comment (rvanlaar) [reply engines/director/lingo/lingo-code.cpp]: This array: ["one", "two", "three"] is printed as: WARNING: ARRAY: [one, two, three]! 
Without the `->c_str()` part its: WARNING: ARRAY: !

The `toString()` works correctly when showing the debug stack after message: 
Stack after: <[one, two, three]>

What would you like to see? A re-implementation of toString() for Arrays, or a custome print function in this code?

File: engines/director/lingo/lingo-codegen.cpp (hunk 2 above)
Comment (rvanlaar) [reply engines/director/lingo/lingo-codegen.cpp]: Pass by reference works.

File: engines/director/lingo/lingo-code.cpp (hunk 1 above)
Comment (sev-) [reply engines/director/lingo/lingo-code.cpp]: Sorry for the created cofusion. I meant:
```
warning("%s", d.toString()->c_str());
```

**Comment by sev-** [comment]
Thanks!

**Pull request merged**